### PR TITLE
Test canned pilot bundles (fixes #28)

### LINKED ISSUE #28: Test canned pilot bundles
@simonjupp added [pilot bundle examples](https://github.com/HumanCellAtlas/metadata-schema/tree/01d279d5c7e46d7b7ddd364dad0df224716a40a4/examples/bundles/public-beta) and suggested that we test the metadata-api against these examples.

### PR DESCRIPTION
**Pull request by hannes-ucsc**

### CODE REVIEW

File: test/humancellatlas/data/metadata/helpers/schema_examples.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                                         'content-type': 'application/json',
+                                         'indexed': True,
+                                         'name': member_path,
+                                         'size': member.size,
+                                         'url': 'bogus',  # FIXME
+                                         'uuid': str(uuid.uuid4()),  # FIXME
+                                         'version': '1'})  # FIXME
+                        schema_name, _, suffix = member_path[:-5].rpartition('_')
+                        if schema_name.endswith('_file') or schema_name == 'file':
+                            manifest.append({**sink.get_checksums(),  # FIXME
```
Comment (hannes-ucsc) [inline test/humancellatlas/data/metadata/helpers/schema_examples.py]: Adress FIXMEs and explain the second manifest entry.

File: test/test.py [hunk 2]
```diff
@@ -29,6 +31,27 @@ def setUp(self):
...
+            ('HPSI_human_cerebral_organoids', AgeRange(min=1419120000, max=1545264000), True),
+            ('Mouse Melanoma', AgeRange(min=3628800, max=7257600), True),
+            ('Single cell transcriptome analysis of human pancreas', AgeRange(min=662256000, max=662256000), True),
+            ('Tissue stability', AgeRange(min=1734480000, max=1892160000), False),
+            ('HPSI_human_cerebral_organoids', AgeRange(min=1419120000, max=1545264000), True),
+            ('1M Immune Cells', AgeRange(min=1639872000, max=1639872000), True)
+        ]:
+            with self.subTest(dir=directory):
+                manifest, metadata_files = download_example_bundle(repo='HumanCellAtlas/metadata-schema',
+                                                                   branch='hs-rename-beta-example-file-names-2',
```
Comment (hannes-ucsc) [inline test/test.py]: Change to `develop` once https://github.com/HumanCellAtlas/metadata-schema/pull/564 lands.

File: src/humancellatlas/data/metadata/helpers/json.py [hunk 3]
```diff
@@ -3,7 +3,7 @@
 
 from dataclasses import field, fields, is_dataclass
 
-from humancellatlas.data.metadata import Entity
```
Comment (rexwangcc) [inline src/humancellatlas/data/metadata/helpers/json.py]: I guess the api change requires us changing the import on Green Box at the same time right? https://github.com/HumanCellAtlas/pipeline-tools/blob/0f04daf7850167fab2b8968ddafaaeaf48d8bd7d/pipeline_tools/input_utils.py#L4

File: test/humancellatlas/data/metadata/helpers/schema_examples.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+                    member_path = member_path[len(path):]
+                    if member_path.endswith('.json'):
+                        with tf.extractfile(member) as tfm:
+                            contents = tfm.read()
+                        json_contents = json.loads(contents)
+                        metadata_files[member_path] = json_contents
+                        sink = ChecksummingSink(64 * 2 ** 20)
+                        sink.write(contents)
+                        manifest.append({**sink.get_checksums(),
+                                         'content-type': 'application/json',
```
Comment (rexwangcc) [inline test/humancellatlas/data/metadata/helpers/schema_examples.py]: Would it matter to add `'content-type': 'application/json; dcp-type="metadata/xxx"'` here?

**Comment by rexwangcc** [comment]
As a side note, I highly encourage to avoid using slashes in the branch name .e.g `rex/my/new/feature`.  It doesn't matter if there are no Jupyter notebooks in the repo, but in our case here, it will prevent the Github from rendering the Jupyter Notebooks properly (I guess this is an issue with the nbviwer rendering service, will submit an issue to the Github). 

To reproduce this, you can compare https://github.com/HumanCellAtlas/metadata-api/blob/feature/rex/do-not-merge/examples/comparison_between_old_and_new_bundle_structures.ipynb(It is not working) and https://github.com/HumanCellAtlas/metadata-api/blob/feature%2Frex/do-not-merge/examples/comparison_between_old_and_new_bundle_structures.ipynb (I escaped the slashes manually in the url)

File: test/humancellatlas/data/metadata/helpers/schema_examples.py (hunk 4 above)
Comment (hannes-ucsc) [reply test/humancellatlas/data/metadata/helpers/schema_examples.py]: Yes, good catch.

**Comment by hannes-ucsc** [comment]
But I love the slashes! They show up like this in SourceTree:

screen shot 2018-09-21 at 11 40 31 pm

File: src/humancellatlas/data/metadata/helpers/json.py (hunk 3 above)
Comment (hannes-ucsc) [reply src/humancellatlas/data/metadata/helpers/json.py]: Oh, yes, I meant to mention this to you. It was a mistake to put everything into `…/metadata/__init__.py`

File: test/test.py (hunk 2 above)
Comment (malloryfreeberg) [reply test/test.py]: @hannes-ucsc HumanCellAtlas/metadata-schema#564 has been merged! Thanks for the catch!

**Pull request merged**